shared/js/content-scope/protections.js

```diff
@@ -34,13 +35,18 @@ export async function loadProtections () {
     ]
 
     for (const protectionName of protectionNames) {
-        const protection = import(`./${protectionName}-protection.js`).then(({ init, load, update }) => {
-            if (load) {
-                load()
-            }
-            return { protectionName, init, update }
-        })
-        protections.push(protection)
+        try {
+            const filename = protectionName.replace(/([a-zA-Z])(?=[A-Z0-9])/g, '$1-').toLowerCase()
+            const protection = import(`./${filename}-protection.js`).then(({ init, load, update }) => {
+                if (load) {
+                    load()
+                }
+                return { protectionName, init, update }
+            })
+            protections.push(protection)
+        } catch (e) {
+            console.error(`Error loading ${protectionName}: ${e}`)
+        }
```

Comment:
@kdzwinel requested a try/catch here. If it failed it would just continue on to the next protection.

Reply:
We could split this out into another issue perhaps.

I'd prefer we failed hard at build time given that the hardcoded list of features are above and the code is all in the repo. We actually rely on the rollup dynamic import to turn this into a normal function call and bundle it all together; we could make a PR/fork that to fail hard with the desired configuration options specified to rollup.

As for catching errors that's a whole discussion:
- For loading the protections, I agree we should probably fail silently to minimise breakage in unexpected scenarios.
- Internally in the protections themselves we should be throwing as the API's do to prevent easy detection of our code.

I'm unsure if the cookie code which loads in the `load()` phase needs to throw exceptions like this; perhaps you both can answer that.

Reply:
Sorry @SlayterDev for confusion, maybe I've put my comment in a wrong place. +1 to what Sam and Jonathan are saying - we should fail hard if there are no external variables here, but when there are external variables (remote config) that can potentially have unknown/missing entries we should fail gracefully only disabling features that we don't have a config for.